### ISSUE TITLE
Centos6.8 with Devtools2 runtime error

### ISSUE THREAD

**Issue by amit777** [grpc/grpc#7903]
I get the following stack track when using the grpc module when running an app:

```
Error: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by /opt/app/node_modules/grpc/src/node/extension_binary/grpc_node.node)
```

Here is the output of:

```
strings /usr/lib64/libstdc++.so.6 | grep GLIBCXX
GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.2
GLIBCXX_3.4.3
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
```

gcc version 4.8.2 20140120 (Red Hat 4.8.2-15) (GCC)

This does not happen on my centos 7 machine

**Comment by amit777** [comment]
More info:
"grpc": "^1.0.0"

```
//simple test
var grpc = require('grpc').
```

just running this causes teh error.

**Comment by soltanmm-google** [comment]
The question is: how old do we want to make our build machines?

**Comment by murgatroid99** [comment]
@nicolasnoble I think @soltanmm-google's question is something that we should decide: when building precompiled binaries, how old should the libc version that we are building against be?

**Comment by nicolasnoble** [comment]
So that's not about libc, that's about libstdc++. We've already done a bunch of work to get older versions of libc running nicely, but libstdc++ is another ball of yarn to untangle.

Right now we're running our build system with Debian / Jessie, and using gcc-4.9. However, we don't want to downgrade that compiler because it has optimizations good enough for us.

The rootcause for your failure is that the node extension uses std::list, which has optimizations in recent versions of libstdc++, that you don't have in CentOS6. The solution might be as simple as stopping using std::list in the node extension, but maybe something else will fail after this.

In all cases, if you compile the node extension, it should work. There should be a `--build-from-source` flag when installing that does that.

**Comment by bayareahank** [comment]
I ran into the exact same issue on CentOS 6.8. BTW, option --build-from-source would fail for devtool set 2~3, the build will only succeed if you use devtool set 4. Still, it would fail when you run your app after compile. 

The only working solution I found out so far is to build an up-to-date version of gcc, then use that to replace the existing old libstdc++.so library. See the link: 
http://unix.stackexchange.com/questions/98633/glibcxx-3-4-15-on-centos-6
